libs/types/src/schema.ts

```diff
@@ -26,50 +18,84 @@ export const Id = z
     (id) => /^[-_a-z\d]+$/i.test(id),
     'IDs may only contain letters, numbers, dashes, and underscores'
   )
+export const HexString: z.ZodSchema<string> = z
+  .string()
+  .nonempty()
+  .refine(
+    (hex) => /^[0-9a-f]*$/i.test(hex),
+    'hex strings must contain only 0-9 and a-f'
+  )
+export const ISO8601Date = z
+  .string()
+  .refine(check8601, 'dates must be in ISO8601 format')
+
+function* findDuplicateIds<T extends { id: unknown }>(
+  identifiables: Iterable<T>
+): Generator<[number, T['id']]> {
+  const knownIds = new Set<T['id']>()
+
+  for (const [index, { id }] of [...identifiables].entries()) {
+    if (knownIds.has(id)) {
+      yield [index, id]
+    } else {
+      knownIds.add(id)
+    }
+  }
+}
 
 // Candidates
-export const Candidate = z.object({
+export const Candidate: z.ZodSchema<t.Candidate> = z.object({
```

Comment:
One of the main changes in this file is to explicitly annotate the types of the schemas. That helps us keep the types in sync (TS will complain if they diverge) and helps TS show better types for parsed values (i.e. `Election` instead of `{ title: string, … }`.